src/Hng.Web/Controllers/ProductController.cs

```diff
@@ -73,5 +72,29 @@ public async Task<ActionResult<IEnumerable<CategoryDto>>> GeProductCategories()
                 categories
             });
         }
+
+        /// <summary>
+        /// Edit user products with update timestamp
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="updateProductDto"></param>
+        /// <returns></returns>
+        [HttpPut("{id:guid}")]
+        [Authorize]
+        [ProducesResponseType(typeof(ProductDto), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> UpdateProduct(Guid id, [FromBody] UpdateProductDto updateProductDto)
+        {
+            var command = new UpdateProductCommand(id, updateProductDto);
+            var result = await _mediator.Send(command);
+            return result != null
+                ? Ok(result)
+                : NotFound(new
+                {
+                    status_code = 404,
```

Comment:
There is a general response class to return responses. Check the Hng.Application/Shared folder for it.

Reply:
Will merge now, next or will have fix for correct response body